src/functions/TestDrive.ps1

```diff
@@ -24,39 +43,76 @@
                     TestDrivePath    = $testDrivePath
                 })
         }
-    } -EachBlockTearDownEnd {
+    }
+
+    $p.EachBlockTearDownEnd = {
         param($Context)
 
+        # Remap drive and variable if missing/wrong? Ex. if nested run was cancelled and didn't re-attach this drive
+        if (-not (& $script:SafeCommands['Test-Path'] TestDrive:\)) {
+            New-TestDrive -Path $Context.Block.PluginData.TestDrive.TestDrivePath
+        }
+
         if ($Context.Block.IsRoot) {
             # this is top-level block remove test drive
             Remove-TestDrive -TestDrivePath $Context.Block.PluginData.TestDrive.TestDrivePath
         }
         else {
-            Clear-TestDrive -TestDrivePath $Context.Block.PluginData.TestDrive.TestDrivePath -Exclude ( $Context.Block.PluginData.TestDrive.TestDriveContent )
+            Clear-TestDrive -TestDrivePath $Context.Block.PluginData.TestDrive.TestDrivePath -Exclude ($Context.Block.PluginData.TestDrive.TestDriveContent)
         }
     }
+
+    $p.End = {
+        param($Context)
+
+        if ($Context.GlobalPluginData.TestDrive.ExistingTestDrivePath) {
+            # If nested run, reattach previous TestDrive PSDrive and variable
+            New-TestDrive -Path $Context.GlobalPluginData.TestDrive.ExistingTestDrivePath
+        }
+    }
+
+    New-PluginObject @p
 }
 
-function New-TestDrive ([Switch]$PassThru, [string] $Path) {
-    $directory = New-RandomTempDirectory
-    $DriveName = "TestDrive"
-    $null = & $SafeCommands['New-PSDrive'] -Name $DriveName -PSProvider FileSystem -Root $directory -Scope Global -Description "Pester test drive"
+function New-TestDrive {
+    param(
+        [string] $Path
+    )
 
-    #publish the global TestDrive variable used in few places within the module
-    if (-not (& $SafeCommands['Test-Path'] "Variable:Global:$DriveName")) {
-        & $SafeCommands['New-Variable'] -Name $DriveName -Scope Global -Value $directory
```

Comment:
Was this intentional to not steal a user variable?
If so, I can revert/change to only override a Pester-created value based on value.

Reply:
I don't remember the history, maybe it was originally not mapping TestDrive to a random directory, and inner Pester simply did not map it's own TestDrive, but instead inherited it?

Reply:
Looks like it's been there since v2, probably to avoid nested Describe-blocks (which I assume was supported) to overwrite the TestDrive. Similar check for existing PSDrive. Should be safe to remove now.